src/openrct2/util/util.h

```diff
@@ -56,4 +56,12 @@ uint32 util_rand();
 uint8 *util_zlib_deflate(uint8 *data, size_t data_in_size, size_t *data_out_size);
 uint8 *util_zlib_inflate(uint8 *data, size_t data_in_size, size_t *data_out_size);
 
+#define add_clamp(value, value_to_add, min_cap, max_cap) \
+	if ((max_cap - abs(value_to_add)) < abs(value) && (value < 0) == (value_to_add < 0)) { \
```

Comment:
are you sure this is correct? the first line seems to assume that min_cap == -max_cap (which is usually not true)